packages/datastore/src/datastore.ts

```diff
@@ -142,6 +245,37 @@ class Datastore implements IIterable<Table> {
   redo(transactionId: string): Promise<void> {
     throw '';
   }
+
+  /**
+   * An optional handler for broadcasting transactions to peers.
+   */
+  transactionBroadcastHandler: IMessageHandler | null = null;
```

Comment:
I would make this a required constructor arg and make it `readonly`.

At this point, you probably want to define a `Datastore.IOptions` interface for passing args to the constructor (or static `create` method).

You may also want to define an object/option for creating the transaction ids, instead of relying solely on the `createDuplexId` function.